lib/ws_servers/api/algos/algo_worker.js

```diff
@@ -157,18 +159,32 @@ class AlgoWorker {
   }
 
   getActiveAlgosFromHost () {
-    return this.host.getAOInstances()
-      .filter((aoInstance) => {
-        const { state = {} } = aoInstance
-        const { active } = state
-        return active
-      })
+    return this.host.getAOInstances().filter((aoInstance) => {
+      const { state = {} } = aoInstance
+      const { active } = state
+      return active
+    })
+  }
+
+  async closeActiveAlgosOnHost () {
+    const promises = []
+    this.host.getAOInstances().forEach((aoInstance) => {
+      const { state = {} } = aoInstance
+      const { gid, active } = state
+      if (!active) {
+        return
+      }
+      promises.push(this.cancelOrder(gid))
+    })
+    await Promise.all(promises)
   }
 
   /**
    * @public
    */
-  close () {
+  async close () {
+    await this.closeActiveAlgosOnHost()
```

Comment:
This will always cancel the order even if we close the honey app normally and we don't want that. Also, it is not necessary to cancel the orders I think. Once, the connection is closed, orders are automatically canceled based on their preferred dms setting. Imo, we just need to clear the algo orders from the table.